mne_bids/write.py

```diff
@@ -1317,29 +1355,63 @@ def write_anat(bids_root, subject, t1w, session=None, acquisition=None,
                                       suffix='T1w.nii.gz')
 
     # Check if we have necessary conditions for writing a sidecar JSON
-    if trans is not None:
-
-        # get trans and ensure it is from head to MRI
-        trans, _ = _get_trans(trans, fro='head', to='mri')
-
-        if not isinstance(raw, BaseRaw):
-            raise ValueError('`raw` must be specified if `trans` is not None')
-
-        # Prepare to write the sidecar JSON
-        # extract MEG landmarks
-        coords_dict = _extract_landmarks(raw.info['dig'])
-        meg_landmarks = np.asarray((coords_dict['LPA'],
-                                    coords_dict['NAS'],
-                                    coords_dict['RPA']))
-
-        # Transform MEG landmarks into MRI space, adjust units by * 1e3
-        mri_landmarks = apply_trans(trans, meg_landmarks, move=True) * 1e3
-
-        # Get landmarks in voxel space, using the mgh version of our T1 data
+    if trans is not None or landmarks is not None:
         t1_mgh = nib.MGHImage(t1w.dataobj, t1w.affine)
-        vox2ras_tkr = t1_mgh.header.get_vox2ras_tkr()
-        ras2vox_tkr = np.linalg.inv(vox2ras_tkr)
-        mri_landmarks = apply_trans(ras2vox_tkr, mri_landmarks)  # in vox
+
+        if landmarks is not None:
+            if raw is not None and verbose:
+                warn('Using provided `landmarks` instead of using ' +
+                     'the `trans` to convert the landmarks from `raw` ' +
+                     'space')
+            if isinstance(landmarks, str):
+                landmarks, coord_frame = read_fiducials(landmarks)
+                landmarks = np.array([landmark['r'] for landmark in
+                                      landmarks], dtype=float)  # unpack
+            else:
+                coords_dict = _extract_landmarks(landmarks.dig)
+                coord_frame = landmarks.dig[0]['coord_frame']
+                if any([landmarks.dig[i]['coord_frame'] != coord_frame
+                        for i in range(3)]):
+                    raise ValueError('Mismatched coordinates: coord_frame, ' +
+                                     '%s, found %s' % (coord_frame,
+                                                       landmarks.dig))
+                landmarks = np.asarray((coords_dict['LPA'],
+                                        coords_dict['NAS'],
+                                        coords_dict['RPA']))
```

Comment:
you could use this https://github.com/mne-tools/mne-python/blob/master/mne/io/_digitization.py#L215